include/libhal/percentage.hpp

```diff
@@ -21,6 +23,28 @@ class percentage
   constexpr static float_t min = -1.0;
   constexpr static float_t zero = 0.0;
 
+  constexpr static bool nearly_equal(percentage p_percentage1,
+                                     percentage p_percentage2,
+                                     float_t p_epsilon)
+  {
+    // get absolute values
+    float_t percentage1_abs = std::abs(p_percentage1.value());
+    float_t percentage2_abs = std::abs(p_percentage2.value());
+    float_t diff = std::abs(p_percentage1.value() - p_percentage2.value());
+
+    if (p_percentage1.value() == p_percentage2.value()) {
+      return true;
+    } else if (p_percentage1.value() == 0 || p_percentage2.value() == 0 ||
+               (percentage1_abs + percentage2_abs <
```

Comment:
```suggestion
               (percentage1_abs + percentage2_abs <
```
This expression is used calculated twice in this code. Notice how we have to calculate this to figure out if this is the correct branch to take. But if it is not, we have to calculate it again in the relative error case. So it would be better if we calculated the sub of the absolute values once and use the result twice.